azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.metrics;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+
+/**
+ * This singleton class CommonMetrics is in charge of collecting varieties of metrics
+ * which are accessed in both web and exec modules. That said, these metrics will be
+ * exposed in both Web server and executor.
+ */
+public enum CommonMetrics {
```

Comment:
Enum is a common practice to construct singleton. Enum can not be inherited, which is one of  the most important advantages.